cmd/yesiscan/main.go

```diff
@@ -468,6 +573,80 @@ func GetConfig(configPath string) (*Config, error) {
 	return &configData, nil
 }
 
+// GetConfigPath returns the expected path to the main config.json file given
+// the input arg for that setting.
+func GetConfigPath(configPath string) (string, error) {
+	// If config path is set, we look in there for a config, otherwise we
+	// use the default xdg path.
+	if configPath != "" {
+		return configPath, nil
+	}
+
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", errwrap.Wrapf(err, "error finding home directory")
+	}
+	if home == "" {
+		return "", fmt.Errorf("home directory is empty")
+	}
+
+	// TODO: get program from an input var perhaps?
+	p := filepath.Join(home, ".config/", program+"/", ConfigFileName)
+	return filepath.Clean(p), nil
+}
+
+// DownloadConfig pulls a config from a magic URI and returns the contents.
+func DownloadConfig(uri string) ([]byte, error) {
```

Comment:
> "Download" to me means that it's being stored on disk

I looked in two dictionaries. Both seemed to disagree. For example: https://www.merriam-webster.com/dictionary/download

Merriam Webster says:

> an act or instance of transferring something (such as data or files) from a usually large computer to the memory of another device (such as a smaller computer)

We are indeed downloading and to the _memory_ of the other computer. Nothing there insists that it is a disk. That is exactly what we're doing, and the function docstring says the same thing:

>  // DownloadConfig pulls a config from a magic URI and returns the contents.

As a result, I don't think this suggested change is valid so I am resolving this conversation.

Reply:
English majors who write dictionaries don't appreciate nuanced differences that can confuse programmers. In technology "Download" can also mean "upload firmware to a device" which is of course unfortunate, but isn't in the dictionary. As a contributor, I was confused by the use of the common vernacular term instead of the HTTP verb. By not using the HTTP verb, you communicate that you mean something different from GET, so I guess you might mean get and then save to disk, and I was surprised when it turned out you meant GET.

Since we're dealing here with HTTP semantics, and programmers who understand the HTTP protocol, it's better to use the relevant HTTP verbs. The word "Download" does not appear in the HTTP specs (or even the FTP specs). The word does, however, appear on websites, where it always implies saving the content to disk.